lua/ui/lobby/changelog/ChangelogDialog.lua

```diff
@@ -245,65 +268,129 @@ Changelog = ClassUI(Group) {
     end,
 
     --- Populates the dialog with the given patch
-    ---@param self UIChangelog
+    ---@param self UIChangelogDialog
     ---@param index number
     PopulateWithPatch = function(self, index)
-        local patch = data.gamePatches[index + 1]
+        ---@type UIChangelogMetadata
+        local changelog = ChangelogOverview.Overview.Changelogs[index + 1]
 
-        if patch then
+        ---@type UIChangelogData
+        local patch = import(changelog.Path).Changelog
 
-            if patch.hasPrettyGithubRelease then
-                self.FooterGithubButton:Enable()
-                self.FooterGithubButton.OnClick = function()
-                    OpenURL(string.format('http://github.com/FAForever/fa/releases/tag/%d', patch.version))
-                end
-            else
-                self.FooterGithubButton:Disable()
-            end
+        if patch then
 
-            if patch.hasPrettyPatchnotes then
-                self.FooterPatchNotesButton:Enable()
-                self.FooterPatchNotesButton.OnClick = function()
-                    OpenURL('http://patchnotes.faforever.com')
+            if changelog.URL then
+                self.FooterOnlineButton:Enable()
+                self.FooterOnlineButton.OnClick = function()
+                    OpenURL(changelog.URL)
                 end
             else
-                self.FooterPatchNotesButton:Disable()
+                self.FooterOnlineButton:Disable()
             end
 
             self.ContentPatchesList:SetSelection(index)
-            self.HeaderSubtitle:SetText(patch.name)
+            self.HeaderSubtitle:SetText(patch.Name or "Unknown name")
```

Comment:
There's a bug where it is always showing "unknown name".
```suggestion
            self.HeaderSubtitle:SetText(changelog.Name or "Unknown name")
```